src/nwb_conversion_tools/tools/data_transfers.py

```diff
@@ -61,14 +63,143 @@ def get_globus_dataset_content_sizes(
+
+        It is recommended to transfer the largest file(s) with minimal batching,
+        and to batch a large number of very small files together.
+
+        It is also generally recommended to submit up to 3 simultaneous transfer,
+        *i.e.*, `source_files` is recommended to have at most 3 items all of similar total byte size.
+    destination_id : string
+        Destination Globus ID.
+    destination_folder : FolderPathType
+        Absolute path to a local folder where all content will be transfered to.
+    display_progress : bool, optional
+        Whether or not to display the transfer as progress bars using `tqdm`.
+        Defaults to True.
+    progress_update_rate : float, optional
+        How frequently (in seconds) to update the progress bar display tracking the data transfer.
+        Defaults to 30 seconds.
+    progress_update_tiemout : float, optional
+        Maximum amount of time to monitor the transfer progress.
+        You may wish to set this to be longer when transferring very large files.
+        Defaults to 10 minutes.
+
+    Returns
+    -------
+    success : bool
+        If 'display_progress'=False, and the transfer was succesfully initiated, then this function returns True.
+
+        If 'display_progress'=True (the default), then this function returns the total status of all transfers
+        when they either finish or the progress tracking times out.
+    task_ids : list of strings
+        List of the task IDs submitted to globus, if further information is needed to reestablish tracking or terminate.
+    """
+    source_files = [[source_files]] if isinstance(source_files, str) else source_files
+    # assertion check is ensure the logical iteration does not occur over the individual string values
+    assert (
+        isinstance(source_files, list) and source_files and isinstance(source_files[0], list) and source_files[0]
+    ), "'source_files' must be a non-empty nested list-of-lists to indicate batched transfers!"
+    assert destination_folder.is_absolute(), (
+        "The 'destination_folder' must be an absolute path to resolve ambiguity with relative Globus head "
+        "as well as avoiding Globus access permissions to a personal relative path!"
+    )
+
+    folder_content_sizes = dict()
+    task_total_sizes = dict()
+    for j, batched_source_files in enumerate(source_files):
+        paths_file = Path(destination_folder) / f"paths_{j}.txt"
+        # .as_posix() to ensure correct string form Globus expects
+        # ':' replacement for Windows drives
+        source_folder = Path(batched_source_files[0]).parent.as_posix().replace(":", "")
+        destination_folder_name = Path(destination_folder).as_posix().replace(":", "")
+        with open(file=paths_file, mode="w") as f:
+            for source_file in batched_source_files:
+                file_name = Path(source_file).name
+                f.write(f"{file_name} {file_name}\n")
```

Comment:
The easiest way to pass files in is to create an intermediate text file?

Reply:
The *only* way to use the `--batch` option, regrettably. https://github.com/globus/globus-cli/blob/main/src/globus_cli/commands/transfer.py#L155-L165

Could technically go down to that lower level of the `globus_sdk` library and add the items programmatically instead, but that requires a heck of a lot of other steps to get unpacked and setup before doing that (auth/client/every other option). Would be much harder to maintain on our end.